src/CMakeLists.txt

```diff
@@ -17,12 +17,20 @@ if (GUI)
     if (WIN32)
         list (APPEND QBT_QT_COMPONENTS WinExtras)
     endif(WIN32)
+    if (APPLE)
+        list (APPEND QBT_GUI_OPTIONAL_LINK_LIBRARIES objc)
+        list (APPEND QBT_QT_COMPONENTS MacExtras)
+    endif (APPLE)
 endif (GUI)
 if (DBUS)
     list (APPEND QBT_QT_COMPONENTS DBus)
 endif (DBUS)
 find_package(Qt5 5.5.1 COMPONENTS ${QBT_QT_COMPONENTS} REQUIRED)
 
+if (GUI AND APPLE)
+    # Fix MOC inability to detect macOS
+    add_definitions(-DQ_OS_MAC)
```

Comment:
I suppose Qt team fixed the bug for qmake actually (which could be the reason they closed the issue), or your custm Qt build has patches from HomeBrew. It certainly fails with cmake though.

The flag is specific for the header, so there are many cpp files. It is saner to define it for all the files, since this flag is defined everywhere but in MOC for all macOS sources, and not defining it may lead to undefined behaviour in other places.

If you think a different comment is more descriptive, would you suggest anything? Maybe adding some Qt bug links?